Dockerfile

```diff
@@ -1,30 +1,50 @@
-FROM alpine:3.15 as certs
-
-RUN apk --update add ca-certificates
-
-# builder
-FROM golang:1.17-alpine3.15 as build
-
-WORKDIR /go/src/app
-RUN apk --update add make git
+# syntax=docker/dockerfile:1.1-experimental
```

Comment:
This is the same Dockerfile that we use in the cluster-api project, the use of distroless/static:nonroot will allow us to avoid copying the cert files from the alpine image which greatly improves caching.

Reply:
Does it really make much of a difference?

Reply:
From a caching perspective, yes, since the `apk --update` commands invalidate the cache at the very beginning of the build. It took me a bit to figure out why I wasn't seeing any benefits of caching after adding the .dockerignore and verifying that the build context was unchanged between runs 😬

Reply:
Well, if it works it works.